packages/extension-store-canner/src/lib/canner/persistenceStore.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  APISchema,
+  ArtifactBuilderOptions,
+  PersistentStore,
+  TYPES,
+  VulcanExtensionId,
+  VulcanInternalExtension,
+} from '@vulcan-sql/core';
+import { inject } from 'inversify';
+import * as oas3 from 'openapi3-ts';
+import { createStorageService } from '../storageService';
+import { BaseStorageService } from '@canner/canner-storage';
+import { CannerStoreConfig, getEnvConfig } from '../config';
+import { geIndicatorFilesOfWorkspaces } from './utils';
+import { ArtifactIndicator } from './models';
+
+export interface RawBuiltInArtifact {
+  // key is source name, value is combined sql js executable code
+  templates: Record<string, string>;
+  schemas: APISchema[];
+  // specs content may be different according to the different spec generator, so we use the record type
+  specs: Record<string, any>;
+}
+
+export type BuiltInArtifact = RawBuiltInArtifact & {
+  specs: {
+    oas3: oas3.OpenAPIObject;
+  };
+};
+
+// key is workspaceSqlName, value is artifact buffer content
+interface WorkspaceArtifact {
+  workspaceSqlName: string;
+  artifact: RawBuiltInArtifact;
+}
+/**
+ * Used the string to identify the extension Id not by the enum "ArtifactBuilderProviderType" because the enum is define in the "core" package,
+ * and the canner extensions is only used for Canner integration, so we could add the "Canner" type in the "ArtifactBuilderProviderType",
+ * If we use the create an other enum and union "ArtifactBuilderProviderType", the enum is only has "Canner" type, so it seems define a other enum is unnecessary,
+ *  */
+@VulcanInternalExtension()
+@VulcanExtensionId('Canner')
```

Comment:
Not correct like you said actually, The `VulcanExtensionId` should be unique when you use the `inversify` factory to create them for the same type Extension, e.g: `TemplateProvider` extension or `DataSource` extension. 

- TemplateProvider => https://github.com/Canner/vulcan-sql/blob/a0711f0d31bbfb69c0f10e14da2361f56e2f258d/packages/core/src/containers/modules/templateEngine.ts#L29
- ProfileReader => https://github.com/Canner/vulcan-sql/blob/develop/packages/core/src/containers/modules/profiles.ts#L19
- DataSource => https://github.com/Canner/vulcan-sql/blob/a0711f0d31bbfb69c0f10e14da2361f56e2f258d/packages/core/src/containers/modules/templateEngine.ts#L29

So even if you use the same type of extension, but you don't use the `inversify` factory, it allows you to duplicate the extension Id, because you could decide how to create them. So that is why our `LocalFileProfileReader` and `LocalStorePersistence` also has the same duplicate `LocalFile` extensionId